### DISCUSSION TITLE
Carplay - search not working, neither GPS

### DISCUSSION THREAD

**Discussion by xerces8** [osmandapp/OsmAnd#12530] [Q&A]
I installed OsmAnd and tried to use it in my car over Carplay and it seems to not work correctly:
 - the map can not be moved by dragging, only by tapping the arrows (in other nav apps dragging works, like AppleMaps, GoogleMaps, Sygic)
 - tapping the the target circle icon to center on my location did nothing
 - search (by POI or partial street name) finds nothing. It says "No results" (or similar, I did not take a screenshot)

I installed the app not long ago, and set to download the map to my country (Slovenia) and the world overview map. Zooming into my country worked fine, streets and details seem OK.

Maybe it needs some time to fully initialize? In any case: it should either work at first, or bring up a message explaining what needs to be done to complete the setup (or whatever is missing, I am guessing here).

At second try (next day when I was driving), the location (target icon) worked. Did not try the search yet.
Between the two tries, when I opened the app on the phone (so no CarPlay), it asked for permission for the location, which I selected "when the app is running")

The about menu says I have version 4.0.6 (the free version).

**Comment by xerces8** [comment]
I'll write this here, as it might be related:

Every time I drive the car, and then leave, my iPhone shows a blue bar at the top, indicating that "OsmAnd Maps is Actively Using Your Location". I never used Carplay during that drive.

Also , in the task list (double press the home button to get the list of "running" apps) there is no trace of OsmAnd.

If I tap the blue bar, OsmAnd appears (both on screen and in the task list), but the map is distorted and can not be moved or zoomed. The rest works (like search, or the menu).
Then I usually kill it by swiping up in the task list.

Apparently CarPlay support is not fully matured yet?

While this happens, the battery drain is normal, so it is more cosmetic than anything.

PS: Originally I wrote about this here: https://apple.stackexchange.com/questions/425921

**Reply by vshcherb** [reply]
Carplay support is in the phase to collect enough feedback and fix it, unfortunately our testing capabilities are limited to interested parties and not all developers have the possibility to see it. So we're going to address all the issues but it might take a long time.
